application/src/main/java/org/thingsboard/server/actors/device/DeviceActorMessageProcessor.java

```diff
@@ -853,6 +858,18 @@ void processCredentialsUpdate(TbActorMsg msg) {
         }
     }
 
+    private void processDeviceTransportSettingsRequest(SessionInfoProto sessionInfo, DeviceTransportSettingsRequestMsg deviceTransportSettingsRequest) {
+        UUID sessionId = getSessionId(sessionInfo);
+        ToTransportMsg msg = ToTransportMsg.newBuilder()
+                .setSessionIdMSB(sessionId.getMostSignificantBits())
+                .setSessionIdLSB(sessionId.getLeastSignificantBits())
+                .setDeviceTransportSettingsMsg(DeviceTransportSettingsMsg.newBuilder()
+                        .setMaxSessionsPerDevice(systemContext.getMaxConcurrentSessionsPerDevice())
```

Comment:
This code is not obvious, and from the first point of view, you forgot to populate most of the structure here. I understand that you hack this structure on the transport level, but this is not clear and misleading.

Please rename this request/response proto and execute it once (lazy load, on first request). So anytime when the gateway requests this information on a transport level, you may provide it without any requests to the core level.